tar/write.c

```diff
@@ -506,6 +506,7 @@ write_archive(struct archive *a, struct bsdtar *bsdtar)
 			if (*arg != '/' && (arg[0] != '@' || arg[1] != '/'))
```

Comment:
It looks like the existing logic here is to consider all args other than those that start with `/` and `@/` as relative, and thus require a chdir. The issue we have in #991 is that `@/absolute/path/to/mtree.file` needs chdir for any relative paths contained within that mtree file. Thus I think you can simplify this to just change the conditional to `if (*arg != '/') do_chdir(bsdtar);`